contracts/adapters/Onboarding.sol

```diff
@@ -192,6 +192,40 @@ contract OnboardingContract is
         dao.sponsorProposal(proposalId, msg.sender);
     }
 
+		function cancelProposal(DaoRegistry dao, uint256 proposalId)
+		    external
+				override
+				onlyMember(dao)
+	  {
+			  require(
+            proposals[address(dao)][proposalId].id == proposalId,
+            "proposal does not exist"
+        );
+
+        IVoting votingContract = IVoting(dao.getAdapterAddress(VOTING));
+				uint256 votingStatus = votingContract.voteResult(dao, proposalId);
+				
+				require(
+					  votingStatus == 0,
+						"proposal cannot be canceled after voting starts"
+				);
+
+        dao.cancelProposal(proposalId);
+
+        ProposalDetails storage proposal = proposals[address(dao)][proposalId];
+
+				if (proposal.token == ETH_TOKEN) {
```

Comment:
storage is great to not load everything in memory. But then if you read a property more than one, it's good to load it in a local variable.

so I would add here
´´´
address token = proposal.token;

if ( token == ETH_TOKEN) {
.... 
} else {
..... use token here
}
```

Reply:
the rule in moloch v2 is
if you cancel the proposal -> return the money to the proposer, not the applicant
You return the money to the applicant only if the proposal failed (another case that we need to cover)

Reply:
It looks like we don't store the proposer address in the proposal yet.  Is it available somewhere I'm missing, or do I need to add it to the stored fields for the proposal?

Reply:
you're right we aren't. We should add it IMO. Just add a new field in ProposalDetails struct and put "msg.sender" as the value